newrelic/hooks/serverless_azure.py

```diff
@@ -0,0 +1,298 @@
+
+    http_request = None
+    for key, value in params.items():
+        if isinstance(value, HttpRequest):
+            http_request = value
+            url_split = urlparse.urlsplit(http_request.url)
+            scheme = url_split.scheme
+            query = url_split.query
+            host, port = url_split.netloc, None if (":" not in url_split.netloc) else url_split.netloc.split(":")
+            break
+
+    # If this is an HTTP http_request object, create a web transaction.
+    # Otherwise, create appropriate type of Transaction in the future
+    if http_request:
+        transaction = WebTransaction(
+            application=application,
+            name=context.function_name,
+            group="AzureFunction",
+            scheme=scheme,
+            host=host,
+            port=port,
+            request_method=http_request.method,
+            request_path=http_request.url,
+            query_string=query,
+            headers=dict(http_request.headers),
+            source=func,
+        )
+
+        # For now, only HTTP triggers are supported
+        trigger_type = "http"
+
+        if hasattr(instance, "_nr_cold_start"):
+            cold_start = True
+            # Delete the attribute so that subsequent calls to this
+            # method are noted as not being cold starts
+            del instance._nr_cold_start
+        else:
+            cold_start = False
+
+        website_owner_name = os.environ.get("WEBSITE_OWNER_NAME", None)
+        if not website_owner_name:
+            subscription_id = "Unknown"
+        else:
+            subscription_id = re.search(r"(?:(?!\+).)*", website_owner_name) and re.search(
+                r"(?:(?!\+).)*", website_owner_name
+            ).group(0)
+        if website_owner_name and website_owner_name.endswith("-Linux"):
+            resource_group_name = re.search(r"\+([a-zA-z0-9\-]+)-[a-zA-Z0-9]+(?:-Linux)", website_owner_name).group(
+                1
+            )
+        elif website_owner_name:
+            resource_group_name = re.search(r"\+([a-zA-z0-9\-]+)-[a-zA-Z0-9]+", website_owner_name).group(1)
+        else:
+            resource_group_name = os.environ.get("WEBSITE_RESOURCE_GROUP", "Unknown")
+        azure_function_app_name = os.environ.get("WEBSITE_SITE_NAME", application.name)
+
+        cloud_resource_id = f"/subscriptions/{subscription_id}/resourceGroups/{resource_group_name}/providers/Microsoft.Web/sites/{azure_function_app_name}/functions/{context.function_name}"
+        faas_name = f"{azure_function_app_name}/{context.function_name}"
+
+        azure_intrinsics = {
+            "cloud.resource_id": cloud_resource_id,
+            "faas.name": faas_name,
+            "faas.trigger": trigger_type,
+            "faas.invocation_id": context.invocation_id,
+        }
+
+        # Only add this attribute if this is a cold start
+        if cold_start:
+            azure_intrinsics.update({"faas.coldStart": True})
+
+        if not transaction:
```

Comment:
Could this exit condition be moved further up before all of the parsing logic to grab data for the intrinsic attributes? To avoid running that code if we don't have a transaction in the first place